examples/user-events/dev/system.clj

```diff
@@ -0,0 +1,66 @@
+(ns system
+  "Functions to start and stop the system, used for interactive
+  development.
+  The `system/start` and `system/stop` functions are required by the
+  `user` namespace and should not be called directly."
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [clojure.java.shell :refer [sh]]
+            [jackdaw.admin :as ja]
+            [user-events.core :as user-events]))
+
+(def system nil)
+
+(defn kafka-admin-client-config
+  []
+  {"bootstrap.servers" "localhost:9092"})
+
+(defn create-topics
+  "Takes a list of topics and creates these using the names given."
+  [topic-config-list]
+  (with-open [client (ja/->AdminClient (kafka-admin-client-config))]
+    (ja/create-topics! client topic-config-list)))
+
+(defn re-delete-topics
+  "Takes an instance of java.util.regex.Pattern and deletes any Kafka
+  topics that match."
+  [re]
+  (with-open [client (ja/->AdminClient (kafka-admin-client-config))]
+    (let [topics-to-delete (->> (ja/list-topics client)
+                                (filter #(re-find re (:topic-name %))))]
+      (ja/delete-topics! client topics-to-delete))))
+
+(defn application-id
+  "Takes an application config and returns an `application.id`."
+  [app-config]
+  (get app-config "application.id"))
+
+(defn destroy-state-stores
+  "Takes an application config and deletes local files associated with
+  internal state."
+  [app-config]
+  (sh "rm" "-rf" (str "/tmp/kafka-streams/" (application-id app-config)))
```

Comment:
Although this is no doubt useful, `rm -rf` is one of those Unix commands which can cause untold misery. I would be inclined to remove this fn from the example code as its going into the public domain ...

Reply:
I agree on this one, I was just following the pattern in the other examples (see word-count). Best have this change applied in other parts as well?

Reply:
@99-not-out How do you suggest we delete the internal state?